src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStoragePublisher.java

```diff
@@ -114,118 +128,138 @@
+
+    private transient final AzureCredentials.StorageAccountCredential storageCreds;
+
     public enum UploadType {
-	INDIVIDUAL,
-	ZIP,
-	BOTH,
-	INVALID;
+        INDIVIDUAL,
+        ZIP,
+        BOTH,
+        INVALID;
     }
 
     @DataBoundConstructor
-    public WAStoragePublisher(final String storageAccName,
-	    final String filesPath, final String excludeFilesPath, final String containerName,
-	    final boolean cntPubAccess, final String virtualPath,
-	    final boolean cleanUpContainer, final boolean allowAnonymousAccess,
-	    final boolean uploadArtifactsOnlyIfSuccessful,
-	    final boolean doNotFailIfArchivingReturnsNothing,
-	    final boolean doNotUploadIndividualFiles,
-	    final boolean uploadZips,
-	    final boolean doNotWaitForPreviousBuild) {
-	super();
-	this.storageAccName = storageAccName;
-	this.filesPath = filesPath.trim();
-	this.excludeFilesPath = excludeFilesPath.trim();
-	this.containerName = containerName.trim();
-	this.cntPubAccess = cntPubAccess;
-	this.virtualPath = virtualPath.trim();
-	this.cleanUpContainer = cleanUpContainer;
-	this.allowAnonymousAccess = allowAnonymousAccess;
-	this.uploadArtifactsOnlyIfSuccessful = uploadArtifactsOnlyIfSuccessful;
-	this.doNotFailIfArchivingReturnsNothing = doNotFailIfArchivingReturnsNothing;
-	this.doNotUploadIndividualFiles = doNotUploadIndividualFiles;
-	this.uploadZips = uploadZips;
-	this.doNotWaitForPreviousBuild = doNotWaitForPreviousBuild;
+    public WAStoragePublisher(final String storageCredentialId,
+            final String filesPath, final String excludeFilesPath, final String containerName,
+            final boolean cntPubAccess, final String virtualPath,
+            final boolean cleanUpContainer, final boolean allowAnonymousAccess,
+            final boolean uploadArtifactsOnlyIfSuccessful,
+            final boolean doNotFailIfArchivingReturnsNothing,
+            final boolean doNotUploadIndividualFiles,
+            final boolean uploadZips,
+            final boolean doNotWaitForPreviousBuild) {
+        super();
+        this.storageCreds = AzureCredentials.getStorageAccountCredential(storageCredentialId);
+        this.filesPath = filesPath.trim();
+        this.excludeFilesPath = excludeFilesPath.trim();
+        this.containerName = containerName.trim();
+        this.cntPubAccess = cntPubAccess;
+        this.virtualPath = virtualPath.trim();
+        this.cleanUpContainer = cleanUpContainer;
+        this.allowAnonymousAccess = allowAnonymousAccess;
+        this.uploadArtifactsOnlyIfSuccessful = uploadArtifactsOnlyIfSuccessful;
+        this.doNotFailIfArchivingReturnsNothing = doNotFailIfArchivingReturnsNothing;
+        this.doNotUploadIndividualFiles = doNotUploadIndividualFiles;
+        this.uploadZips = uploadZips;
+        this.doNotWaitForPreviousBuild = doNotWaitForPreviousBuild;
+        this.storageCredentialId = storageCredentialId;
+        this.storageAccName = this.storageCreds.getStorageAccountName();
```

Comment:
same thing as in Builder. storageAccName should be transient or completely removed (you can get it from the credential)